### ISSUE TITLE
Use BuilderPattern for `Server`

### ISSUE THREAD

**Issue by tusharmath** [zio/zio-http#753]
**Is your feature request related to a problem? Please describe.**
Using the `++` operator is quite cumbersome for set Server settings.

**Describe the solution you'd like**
Using the builder pattern is a much better way specify server settings.

```scala
def app: HttpApp[Any, Nothing] = ???

def server = Server(app, port = 8080)
  .withFlowControl(true)
  .withKeepAlive(false)
  .withLeakDetection(LeakDetection.Paranoid)
  .start
```

**Additional context**
A prototype needs to be done to showcase how the API will look once changed completely

**Comment by jgoday** [comment]
Hi @tusharmath I have created a PR #768, I don't know if it will be the best/correct approach, feel free to remove it if necessary.

**Comment by tusharmath** [comment]
fixed in #768
